packages/components/pie-modal/src/index.ts

```diff
@@ -317,7 +338,9 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
             size="${size}"
             position="${position}"
             ?hasBackButton=${hasBackButton}
+            ?hasActions=${!!leadingAction || !!supportingAction}
```

Comment:
Could you try the below instead.

`?hasActions=${ifDefined(leadingAction || supportingAction)}`

Reply:
<img width="1068" alt="Screenshot 2023-07-25 at 15 43 38" src="https://github.com/justeattakeaway/pie/assets/16766185/3b3a380c-f112-44bc-9272-b3e11e4c3d29">
@kevinrodrigues does this seem right?

Reply:
<img width="1125" alt="Screenshot 2023-07-25 at 15 44 17" src="https://github.com/justeattakeaway/pie/assets/16766185/e0ac6237-719b-47b8-a5e2-bb88c385de5c">

@kevinrodrigues does this seem right?

Reply:
It doesn't look right. On second thoughts and out of interest why do you need the coerson boolean conversion? Doesn't ${leadingAction || supportingAction} work?

Reply:
Removing the `!!` would probably work but I felt this was clearly testing the truthiness and the way you suggested made it look a bit more like the props themselves could be booleans

Reply:
Our props are named so indicate if it is a boolean or not `isFooBar` = Boolean vs `leadingAction` = something else (object). I dont think the conversion should be applied here.

Reply:
Ok, my opinion is that the line I've added is far more explicit. It's telling you that either prop, as long as it is truthy, will meet the criteria for the attribute to be added.

It's all the benefit of `Boolean()` with less code. I think `?hasActions=${leadingAction || supportingAction}` looks as though you're trying to use the value for the attribute.

Reply:
`leadingAction` and `supportingAction ` are optional objects. if they aren't passed, their value is undefined and I believe just checking leadingAction || supportingAction would be enough.

Reply:
tho I would be extra defensive here as regardless if we use Boolean or !!, this will still pass if users pass an empty object `{}` and that's misleading. 

// Boolean({}) == true

I think lodash has a function called isEmpty that we can use for this case, but we don't have lodash added to our web components / alternatively, create a helper  and import it 

https://www.freecodecamp.org/news/check-if-an-object-is-empty-in-javascript/

Reply:
@kevinrodrigues given this has gone on a bit I'm going to take your suggestion. Please can you re-review?

Reply:
Just had another thought.  If leadingAction isn't supplied i.e is `undefined` then `supportingAction` isn't rendered in the dom (even if it is supplied) so how does this check work in that scenario when the modal has no actions rendered but here we are saying `hasActions` gets a value back from either leadingAction or supportingAction?

Reply:
My 2 cents: those expressions not so easy to read could be extracted to a constant. It also makes debugging easier, since you can add a breakpoint or log the expression value.